pykern/pkasyncio.py

```diff
@@ -16,25 +19,91 @@
 _cfg = None
 
 
+class _WebRequestHandler(tornado.web.RequestHandler):
+    async def get(self, *args, **kwargs):
+        # method is argument to request object, which is just self
+        await ReqBase.http_request(self)
+
+    async def post(self, *args, **kwargs):
+        await ReqBase.http_request(self)
+
+    async def put(self, *args, **kwargs):
+        await ReqBase.http_request(self)
+
+    def write_error(self, status_code, *args, **kwargs):
+        if status_code >= 500 and (e := kwargs.get("exc_info")):
+            pkdlog("exception={} stack={}", e[1], pkdexc(e))
+        super().write_error(status_code, *args, **kwargs)
+
+
+class _WebRequestRouter:
+
+    async def get(self, *args, **kwargs):
+        await self._pk_get(self.__authenticate())
+
+    async def post(self, *args, **kwargs):
+        await self._pk_post(self.__authenticate())
+
+    async def put(self, *args, **kwargs):
+        await self._pk_put(self.__authenticate())
+
+    def write_error(self, status_code, *args, **kwargs):
+        if status_code >= 500 and (e := kwargs.get("exc_info")):
+            pkdlog("exception={} stack={}", e[1], pkdexc(e))
+        super().write_error(status_code, *args, **kwargs)
```

Comment:
This class doesn't inherit so I'm not sure where the super() call goes.

Reply:
cd2be9ee